platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java

```diff
@@ -46,17 +54,19 @@
   public static synchronized Mapbox getInstance(@NonNull Context context, @NonNull String accessToken) {
     if (INSTANCE == null) {
       Context appContext = context.getApplicationContext();
-      INSTANCE = new Mapbox(appContext, accessToken, new LocationSource(appContext));
-      LocationEngine locationEngine = new LocationSource(appContext);
-      locationEngine.setPriority(LocationEnginePriority.NO_POWER);
+      List<LocationSourceChain> locationSources = initLocationSources();
+      LocationSourceProvider locationSourceProvider = new LocationSourceProvider(locationSources);
+      LocationEngine locationSource = locationSourceProvider.supply(context);
+      INSTANCE = new Mapbox(appContext, accessToken, locationSource);
+      locationSource.setPriority(LocationEnginePriority.NO_POWER);
       MapboxTelemetry.getInstance().initialize(
-        appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, locationEngine);
+        appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, locationSource);
       ConnectivityReceiver.instance(appContext);
     }
     return INSTANCE;
   }
 
-  Mapbox(@NonNull Context context, @NonNull String accessToken, LocationSource locationSource) {
+  Mapbox(@NonNull Context context, @NonNull String accessToken, LocationEngine locationSource) {
```

Comment:
@zugaldia Problem though is `public` [`getLocationSource()`](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java#L136-L138) which returns a `LocationSource`.